### PR TITLE
NISP-2297: Added cache code for summary API, test for findByNino meth…

### PR DESCRIPTION
**Pull request by prashantchoudhari**
…od and updated test for NpsConnector-getSummary

1. Update NpsSummary - model
2. getSummary using cache - insert and find by nino method.
3. added test for findByNino method
4. updated test for NpsConnector-Summary method.

Remaining
1. Test for read write - @PaulCDurham  is working on it already.
2. Any more tests?

### CODE REVIEW

File: test/uk/gov/hmrc/nationalinsurancerecord/connectors/NpsConnectorSpec.scala [hunk 1]
```diff
@@ -25,83 +25,46 @@ import org.scalatestplus.play.OneAppPerSuite
...
 
     "return a failed Summary when there is an http error and pass on the exception" in {
-      when(connector.http.GET[HttpResponse](Matchers.any())(Matchers.any(), Matchers.any()))
-        .thenReturn(Future.failed(new BadRequestException("Bad Request exception")))
-      ScalaFutures.whenReady(connector.getSummary(generateNino()).failed) { ex =>
-        ex shouldBe a[BadRequestException]
-      }
+      val npsSummaryF = connector.getSummary(generateNino())(HeaderCarrier())
+      await(npsSummaryF) shouldBe testSummaryModel
+      verify(connector.http, never()).GET(Matchers.any())(Matchers.any(), Matchers.any())
```
Comment (dspork) [inline test/uk/gov/hmrc/nationalinsurancerecord/connectors/NpsConnectorSpec.scala]: This test does not match the description

File: test/uk/gov/hmrc/nationalinsurancerecord/connectors/NpsConnectorSpec.scala [hunk 2]
```diff
@@ -25,83 +25,46 @@ import org.scalatestplus.play.OneAppPerSuite
...
-        val npsSummaryF = await(connector.getSummary(nino)(HeaderCarrier()))
-        npsSummaryF.rreToConsider shouldBe false
-        npsSummaryF.finalRelevantYear shouldBe 2016
-        npsSummaryF.dateOfBirth shouldBe new LocalDate(1952,11,21)
-        npsSummaryF.dateOfDeath shouldBe None
-        npsSummaryF.earningsIncludedUpTo shouldBe new LocalDate(2014,4,5)
-    }
-
-    "return a failed Summary when it cannot parse JSON and report validation errors" in {
-      when(connector.http.GET[HttpResponse](Matchers.any())(Matchers.any(), Matchers.any()))
```
Comment (dspork) [inline test/uk/gov/hmrc/nationalinsurancerecord/connectors/NpsConnectorSpec.scala]: We are no longer testing the connector?

**Comment by prashantchoudhari** [comment]
@dspork - please check if we need more tests here and review PR

**Comment by prashantchoudhari** [comment]
logger level changed back to INFO and added test for mongo

**Pull request merged**